src/Form/MediaLibraryEmbeddableForm.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+namespace Drupal\stanford_media\Form;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\media\OEmbed\ResourceFetcherInterface;
+use Drupal\media\OEmbed\UrlResolverInterface;
+use Drupal\media_library\MediaLibraryUiBuilder;
+use Drupal\media_library\OpenerResolverInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Form\FormBuilderInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\media_library\Form\OEmbedForm;
+
+/**
+ * Media library add stanford embed input form.
+ *
+ * @package Drupal\stanford_media\Form
+ */
+class MediaLibraryEmbeddableForm extends OEmbedForm {
+
+  /**
+   * The current User.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $currentUser;
+
+  /**
+   * The name of the oEmbed field.
+   *
+   * @var string
+   */
+  public $oEmbedField;
+
+  /**
+   * The name of the Unstructured field.
+   *
+   * @var string
+   */
+  public $unstructuredField;
+
+  /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * {@inheritDoc}
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, MediaLibraryUiBuilder $library_ui_builder, UrlResolverInterface $url_resolver, ResourceFetcherInterface $resource_fetcher, ConfigFactoryInterface $config_factory, AccountInterface $account, OpenerResolverInterface $opener_resolver = NULL) {
+    parent::__construct($entity_type_manager, $library_ui_builder, $url_resolver, $resource_fetcher, $opener_resolver);
+    $this->currentUser = $account;
+    $this->configFactory = $config_factory;
+
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('media_library.ui_builder'),
+      $container->get('media.oembed.url_resolver'),
+      $container->get('media.oembed.resource_fetcher'),
+      $container->get('config.factory'),
+      $container->get('current_user'),
+      $container->get('media_library.opener_resolver')
+    );
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function getFormId() {
+    return $this->getBaseFormId() . '_embeddable';
+  }
+
+  /**
+   * Sets up the field names for the media type.
+   *
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current form state.
+   */
+  public function setFieldNames(FormStateInterface $form_state) {
+    $media_type_id = $this->getMediaType($form_state)->id();
+
+    $this->oEmbedField = $this->configFactory
```

Comment:
you already are getting the media type above. there's no need to get load the config factory. you can simply use `$this->getMediaType($form_state)->getSource()->getConfiguration()` to get the source configuration. Also you should always load the entity instead of loading from config. It's more reliable and implements the various hooks that are appropriate at those times.